File: 백준/Bronze/32209. 다음 달에 봐요/다음 달에 봐요.c
#include <stdio.h>

int main() {
    int Q, a, b, q = 0;
    
    scanf("%d", &Q);
    
    for (int i = 0; i < Q; i++) {
        scanf("%d %d", &a, &b);
        
        if (a % 2 != 0) {
            q += b;
        } else {
            q -= b;
        }

        if (q < 0) {
            printf("Adios\n");
            return 0;
        }
    }

    printf("See you next month\n");
    return 0;
}

File: 백준/Bronze/32209. 다음 달에 봐요/다음 달에 봐요.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int Q = sc.nextInt();
        int q = 0;

        for (int i = 0; i < Q; i++) {
            int a = sc.nextInt();
            int b = sc.nextInt();

            if (a % 2 != 0) {
                q += b;
            } else {
                q -= b;
            }

            if (q < 0) {
                System.out.println("Adios");
                return;
            }
        }

        System.out.println("See you next month");
    }
}
